### ISSUE TITLE
How do I use Google OR-tools in my Android application?

### ISSUE THREAD

**Issue by samalpys** [google/or-tools#1130]
I am trying to solve a scheduling problem using an OR-tools and integrate this into my project-management android application.

I have added the com.google.ortools.jar on my lib folder inside my android application project and faced the problem with "class file has wrong version 54.0, should be 52.0", which tells me that com.google.ortools.jar is compiled with Java 11, but my android app with Java8 only.

The problem is that android SDK supports only Java 8 and lower, but I somehow manually forced it to use Java 11 provided by Oracle, not Android.

This manipulation led to some other weird problems that I have never seen before and it seems like what I am doing is just wrong.

I wonder, if anyone had integrated OR-tools inside their Android application. 
Is there any proper way of doing it? Can't find any manual in the web

**Comment by Mizux** [comment]
duplicate of #537 

TLDR: we still don't support cross-compilation nor ARM...

**Comment by lperron** [comment]
You cannot easily.  The library contains native code that is not compiled
for arm.
You would need to compile from source, maybe with a cross compiler... A
long endeavor.
Laurent Perron | Operations Research | lperron@google.com | (33) 1 42 68 53
00

**Comment by forwcj517** [comment]
Is it impossible to use the or tools in android ?

**Comment by forwcj517** [comment]
If possible, can you give me a guide?

**Comment by lperron** [comment]
You need to compile the c++ library yourself. It is possible.  We compile
ortools internally for android, but the toolchain is different.
You may need help on files in the ortools/port/.
Laurent Perron | Operations Research | lperron@google.com | (33) 1 42 68 53
00

**Comment by Mizux** [comment]
While or-tools don't use fancy x86 stuff so cose is portable to arm without any change, I won't say  it's 100% guarantee for all our dependencies e.g. CBC/CLP solver, also you have to adapt any build system (bazel, make, cmake) to be able to use a cross-toolchain instead of performing a native build.

note: I would go for cmake bazel build IMHO since android ndk already provide a toolchain IIRC...
note2: it is on our todo list with a very low priority -> no ETA.
